plugins/modules/vmware_vspan_session.py

```diff
@@ -477,6 +477,63 @@ def check_if_session_name_is_free(self):
                 return False
         return True
 
+    def check_source_port_transmitted(self, session):
+        if self.source_port_transmitted is not None:
+            port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(portKey=str(self.source_port_transmitted))
+            if not self.dv_switch.FetchDVPorts(vim.dvs.PortCriteria(portKey=port.portKey)):
+                self.module.fail_json(msg="Couldn't find port: {0:s}".format(self.source_port_transmitted))
+            session.sourcePortTransmitted = port
+
+    def check_source_port_received(self, session):
+        if self.session_type == 'remoteMirrorDest':
+            if self.source_port_received is not None:
+                port = vim.dvs.VmwareDistributedVirtualSwitch.VspanPorts(vlans=[int(self.source_port_received)])
+                if int(self.source_port_received) not in self.dv_switch.QueryUsedVlanIdInDvs():
+                    self.module.fail_json(msg="Couldn't find vlan: {0:s}".format(self.source_port_received))
+                session.sourcePortReceived = port
+        else:
+            if self.source_port_received is not None:
```

Comment:
I would put this `if` one layer above since it's redundant with the previous block.

You can also start the method with:
```
if not self.source_port_received:
    return
```